plugins/synaptics-rmi/fu-synaptics-rmi-ps2-device.c

```diff
@@ -97,6 +97,140 @@ WriteByte (FuSynapticsRmiPs2Device *self, guint8 buf, guint timeout, GError **er
 	return FALSE;
 }
 
+static gboolean
+WriteRMIRegister (FuSynapticsRmiPs2Device *self, 
+				 guint8 addr, 
+				 const guint8 *buf, 
+				 guint8 len, 
+				 guint timeout, 
+				 GError **error)
+{
+
+	if (!self->inRMIBackdoor && !fu_synaptics_rmi_ps2_device_enable_RMI_backdoor(self, error)) {
+		g_prefix_error (error, "failed to enable RMI backdoor: ");
+		return FALSE;
+	}
+		
+
+	if(!WriteByte(self, edpAuxSetScaling2To1, timeout, error)) {
+		g_prefix_error (error, "failed to WriteByte: ");
+		return FALSE;
+	}
+
+	if (!WriteByte(self, edpAuxSetSampleRate, timeout, error)) {
+		g_prefix_error (error, "failed to WriteByte: ");
+		return FALSE;
+	}
+
+	if (!WriteByte(self, addr, timeout, error)) {
+		g_prefix_error (error, "failed to WriteByte: ");
+		return FALSE;
+	}
+
+	for(guint8 numBytes = 0 ; numBytes < len ; numBytes++){
+		if (!WriteByte(self, edpAuxSetSampleRate, timeout, error)) {
+			g_prefix_error (error, "failed to WriteByte: ");
+			return FALSE;
+		}
+
+		if (!WriteByte(self, *(buf + numBytes), timeout, error)) {
+			g_prefix_error (error, "failed to WriteByte: ");
+			return FALSE;
+		}
+	}
+	
+
+	g_usleep (1000*20);
+
+	return TRUE;
+}
+
+static gboolean
+ReadRMIRegister (FuSynapticsRmiPs2Device *self, 
+				guint8 addr, 
+				guint8 * buf,
+				GError **error)
+{
+	if (!buf) {
+		return FALSE;
+	}
```

Comment:
we normally do `== NULL` for pointers, not `!`

Reply:
this seems like it should be `g_return_val_if_fail` actually